anta/tests/stp.py

```diff
@@ -0,0 +1,168 @@
+"""
+Test functions related to various Spanning Tree Protocol (STP) settings
+"""
+from __future__ import annotations
+
+import logging
+from typing import Any, Dict, cast
+
+from anta.models import AntaTest, AntaTestCommand, AntaTestTemplate
+
+logger = logging.getLogger(__name__)
+
+
+class VerifySTPMode(AntaTest):
```

Comment:
This does not work on my setup.
I had a `KeyError` (VLAN provided as input was not defined on the device):
```
           DEBUG    Test VerifySTPMode for device leaf2a: running test                                            models.py:226
           DEBUG    self.instance_commands is: [AntaTestCommand(command='show spanning-tree vlan 100',                stp.py:57
                    version='latest', ofmt='json', output={'spanningTreeVlanInstances': {}},
                    template=AntaTestTemplate(template='show spanning-tree vlan {vlan}', version='latest',
                    ofmt='json'), template_params={'vlan': '100'})]
           DEBUG    dataset for vlan 100 is: {'spanningTreeVlanInstances': {}}                                        stp.py:64
           ERROR    Exception raised for test VerifySTPMode (on device leaf2a) - KeyError ('100')                 models.py:229
           DEBUG    Traceback (most recent call last):                                                            models.py:230
                      File "/mnt/lab/projects/anta/anta/models.py", line 227, in wrapper
                        function(self, **kwargs)
                      File "/mnt/lab/projects/anta/anta/tests/stp.py", line 66, in test
                        if
                    command_output["spanningTreeVlanInstances"][vlan_id]["spanningTreeVlanInstance"]["protocol"]
                    != mode:
                    KeyError: '100'
```